pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1704,6 +1706,57 @@ func (n Nodes) Find(f func(NodeStatus) bool) *NodeStatus {
 	return nil
 }
 
+func (n Nodes) Get(key string) (*NodeStatus, error) {
+	val, ok := n[key]
+	if !ok {
+		return nil, fmt.Errorf("key was not found for %s", key)
+	}
+	return &val, nil
+}
+
+func (n Nodes) Has(key string) bool {
+	_, err := n.Get(key)
+	return err == nil
+}
+
+func (n Nodes) GetPhase(key string) (*NodePhase, error) {
+	val, ok := n[key]
```

Comment:
nit: You can use the `Get` function you implemented here for this and the other times you do `n[key]`